### PR TITLE
feat: As an attestation issuer, I would like to remain compatible with the ERC-721 standard

### LINKED ISSUE #137: [US] As an attestation issuer, I would like to remain compatible with the ERC-721 standard
As an attestation consumer,
I want to be able to check if an address has an attestation using the ERC-721 owner method,
So that I can consume attestations without having to write new code for my dapps

### Description

Many promotions, quests, loyalty programs and gating mechanisms already use NFTs as a standard. If we implement at least part of the erc-721 interface, then all of those dapps will be able to read from Verax without having to write any extra custom logic on their end. From their perspective, they will not be reading "attestations" but will simply be checking if an address has an NFT (which is what they do already).

This would be implemented **on the portal level**, thereby turning that portal into a quasi-SBT.

To do this we would need to provide an example portal that implements _at least_ the following two function:

```Solidity
/// @notice Count all NFTs assigned to an owner
/// @dev NFTs assigned to the zero address are considered invalid, and this
///  function throws for queries about the zero address.
/// @param _owner An address for whom to query the balance
/// @return The number of NFTs owned by `_owner`, possibly zero
function balanceOf(address _owner) external view returns (uint256);

/// @notice Find the owner of an NFT
/// @dev NFTs assigned to zero address are considered invalid, and queries
///  about them do throw.
/// @param _tokenId The identifier for an NFT
/// @return The address of the owner of the NFT
function ownerOf(uint256 _tokenId) external view returns (address);
```

The `ownerOf` function would simply convert the `uint256 _tokenId` param to a `bytes32 attestationId` and retrieve the respective attestation.

The `balanceOf` function would return the number of attestations that a subject has. This would require that the address _owner be cast as bytes and used to look up `mapping (bytes => uint256) numberOfAttestations;` and this would obviously require maintaining this mapping in state.

Ideally these functions would however be implemented on the portal level, as then the portal can be regarded as a NFT by other dapps that don't yet know anything about attestations. They can query the portal directly, and the portal can return the attestation metadata as a tokenURI which it can refer to an indexer (passing the token id) or some off-chain attestation data.

See the [erc-721 specification](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md) for more details.

### PR DESCRIPTION
**Pull request by satyajeetkolhapure**
### Related ticket

Fixes #137 

### Type of change

- [ ] Chore
- [ ] Bug fix
- [x] New feature
- [ ] Documentation update

## Check list

- [x] Unit tests for any smart contract change
- [ ] Contracts and functions are documented

### CODE REVIEW

**Review by alainncls — CHANGES_REQUESTED** [review]

File: test/example/NFTPortal.t.sol [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  function setUp() public {
+    router.initialize();
+    router.updateModuleRegistry(address(moduleRegistryMock));
+    router.updateAttestationRegistry(address(attestationRegistryMock));
+
+    nftPortal = new NFTPortal(modules, address(router));
+  }
+
+  function test_balanceOf_ownerOf() public {
```
[the remark this answers is not in the document]
Comment (satyajeetkolhapure) [reply test/example/NFTPortal.t.sol]: fixed

File: src/example/NFTPortal.sol [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  /** @notice Count all attestations assigned to an owner
+   * @param owner An address for whom to query the balance
+   * @return The number of attestations owned by `owner`, possibly zero
+   */
+  function balanceOf(address owner) public view virtual override returns (uint256) {
+    return numberOfAttestationsPerOwner[abi.encode(owner)];
+  }
+
+  /** @notice Find the owner of an attestation
```
Comment (alainncls) [inline src/example/NFTPortal.sol:31]:
```suggestion
  /** 
   * @notice Find the owner of an attestation
```

File: src/example/NFTPortal.sol [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+ */
+contract NFTPortal is AbstractPortal, ERC721 {
+  mapping(bytes owner => uint256 numberOfAttestations) private numberOfAttestationsPerOwner;
+
+  constructor(
+    address[] memory modules,
+    address router
+  ) AbstractPortal(modules, router) ERC721("NFTPortal", "NFTPortal") {}
+
+  /** @notice Count all attestations assigned to an owner
```
Comment (alainncls) [inline src/example/NFTPortal.sol:23]:
```suggestion
  /** 
   * @notice Count all attestations assigned to an owner
```

**Review by alainncls — CHANGES_REQUESTED** [review]

File: test/example/NFTPortal.t.sol [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    address ownerOfSecondAttestation = nftPortal.ownerOf(2);
+    assertEq(ownerOfFirstAttestation, address(1));
+    assertEq(ownerOfSecondAttestation, address(1));
+  }
+
+  function testSupportsInterface() public {
+    bool isIERC165Supported = nftPortal.supportsInterface(type(IERC165).interfaceId);
+    assertTrue(isIERC165Supported);
+    bool isIERC721Supported = nftPortal.supportsInterface(type(IERC721).interfaceId);
+    assertEq(isIERC721Supported, true);
```
Comment (alainncls) [inline test/example/NFTPortal.t.sol:67]:
```suggestion
    assertTrue(isIERC721Supported)
```

File: test/example/NFTPortal.t.sol [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    bool isEASAbstractPortalSupported = nftPortal.supportsInterface(type(AbstractPortal).interfaceId);
+    assertEq(isEASAbstractPortalSupported, true);
```
Comment (alainncls) [inline test/example/NFTPortal.t.sol:69]:
```suggestion
    assertTrue(isEASAbstractPortalSupported)
```

**Review by alainncls — CHANGES_REQUESTED** [review]

**Review by alainncls — APPROVED** [review]

**Pull request merged**